### PR TITLE
Partially add mocks for unsubscribes endpoints and rewrite unsubscribes test to use them.

### PR DESCRIPTION
**Pull request by hirokisan**
SSIA

It's not perfect, but I hope you're okay with the changes!
If it's an unnecessary change, I'll withdraw it.

### CODE REVIEW

File: unsubscribes_test.go [hunk 1]
```diff
@@ -1,33 +1,31 @@
-package mailgun
+package mailgun_test
```
Comment (tsidei) [inline unsubscribes_test.go:1]: please keep the package name the same.  Only one package should exist per directory and that also makes it inconsistent with the rest of the code

**Review by tsidei — CHANGES_REQUESTED** [review]
It's definitely needed. Thanks for taking the time to contribute!

It all looks good, except the renamed package name. Please revert that change and it's good to go.

Also, do you plan on fully implementing the `unsubscribe` endpoints? Asking just because I was planning to implement them too.

File: unsubscribes_test.go (hunk 1 above)
Comment (vtopc) [reply unsubscribes_test.go:1]: > Only one package should exist per directory

it doesn't relate to `_test`, TBH.

Having `_test` is more about Black box testing - https://medium.com/tech-at-wildlife-studios/testing-golang-code-our-approach-at-wildlife-6f41e489ff36

Comment (vtopc) [reply unsubscribes_test.go:1]: @tsidei, so the question is more about do we need a White box or Black box testing here.

Comment (tsidei) [reply unsubscribes_test.go:1]: oh yeah, that's true. Thanks @vtopc, forgot about that. Nevermind @hirokisan , all good here.

**Review by tsidei — APPROVED** [review]

**Comment by hirokisan** [comment]
I'm glad you accepted the changes.

>do you plan on fully implementing the unsubscribe endpoints?

I don't have any plans at the moment.

>I was planning to implement them too

That's great.

**Pull request merged**